### ISSUE TITLE
[Attention!] For people who use fd and update it to v8.1.0, the results of fzf will show up very slow. Come in for details.

### ISSUE THREAD

**Issue by yanzhang0219** [junegunn/fzf#2059]
- [x] I have read through the manual page (`man fzf`)
- [x] I have the latest version of fzf
- [x] I have searched through the existing issues

## Info

- OS
    - [ ] Linux
    - [x] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [ ] bash
    - [x] zsh
    - [ ] fish
    
## Problem / Steps to reproduce

I use macOS 10.15.5. 

**NOTE**: **There is nothing wrong with fd itseft. fd itself performs very well like before.** fd updated it dependencies which won't affect fd itself, however, this updating brings a side effect only when using it with fzf. For more details, see my post here: https://github.com/sharkdp/fd/issues/599

If you are using `fd` in `fzf`, the searching results will show up very very slow and most of the time they will appear at the end of scanning. If you encountered the same abnormal behavior, just follow and refer to this issue https://github.com/sharkdp/fd/issues/599.

The possible reason is that the upgrading of fd dependencies (see https://github.com/sharkdp/fd/issues/599#issuecomment-633300032). And the author of fd is now investigating it deeply. You can stick to v8.0.0 so far to make fzf work as normal. 

Just let you know. Thank you. 

Update:
It seems that fzf with fd v8.1.1 doesn't have this performance any more. Just give it a shot.

**Comment by LuRsT** [comment]
Is this still a problem @yanzhang0219? I'm using fd version 8.1.1, so I imagine that this issue has been solved by now, if that's the case, we could close this.

**Comment by yanzhang0219** [comment]
I believe it is not solved. You can check the issue page I linked above and a guy said 8.1.1 is still very slow when it is used with fzf. I wrote a very detailed description of the slowness in that issue page. Also, I just asked the author of fd to see whether he has some updates. Please leave this open now. If that issue is solved, I will close this. Thanks.

**Comment by sharkdp** [comment]
To be honest, I don't think this belongs on the `fzf` issue tracker. The linked ticket (https://github.com/sharkdp/fd/issues/599) is about one particular performance regression in one particular use case. Most use cases will still show the same performance. Others will show better performance.

Personally, I have not been able to reproduce the reported regression in benchmarks (see https://github.com/sharkdp/fd/issues/599#issuecomment-640084589).

I'm not saying that the reported effect is not real, but it's a subtle detail in how `fd` works. It should not be discussed here. It's not like there is an inherent problem with fd v8.1.x. It can be used perfectly fine with `fzf`, just like before.

**Comment by yanzhang0219** [comment]
Yes. I opened an issue here just as a reminder in case some users encounter this issue, and I agree that this thread shouldn't be used to discuss the issue itself. So please detour to the specific issue page instead. Thank you!

**Comment by sharkdp** [comment]
It's not just that. By writing things like "the results of fzf *will show up very slow*" with a big "Attention!" marker in the beginning, you make it sound like there is an inherent problem with `fd` 8.1.0.

**Comment by yanzhang0219** [comment]
Okay closing this!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Comment by yanzhang0219** [comment]
I reopen this as a reference for users having this same concern and issue. I think I said very clearly that this performance issue is due to the dependencies of fd, not fd itselt.

**Comment by yanzhang0219** [comment]
I just updated fd to v8.1.1 and it seems that this performance issue doesn't exist. Closing this. Thanks.

**Comment by yanzhang0219** [comment]
As my further investigated, `fzf` with `fd v8.1.1` still bring this performance issue. I put my test procedure below for the users who encounter the same thing as a reference. 

**Note again, this doesn't mean fd has any internal problem. fd itself performs very well as before. This abnormal is just due to the changing of the dependencies of fd, and it only happens when using it with fzf.**

See the GIF below. I run `fzf` on my home directory and tried to search `~/.config/zsh/.zshrc` file. After launching `fzf`, I typed a query like `.config zsh zshrc`. You can see that as I was tying, `~/.config/zsh/.zshrc` file didn't show up, and instead, a zshrc file in my Trash bin appeared whose path is `~/.Trash/dotfiles-tmp/.config/zsh/.zshrc`

![8 1 1](https://user-images.githubusercontent.com/11582667/86560175-9480b180-bf12-11ea-9b86-7076b6c8b22b.gif)

Then next, I uninstalled `fd v8.1.1` and installed `fd v8.0.0` instead. And I performed the same test. This time you can see that as I was typing the query, `~/.config/zsh/.zshrc` file appeared immediately on the first place.

![8 0 0](https://user-images.githubusercontent.com/11582667/86560566-749dbd80-bf13-11ea-8e95-aa469e8e5dd8.gif)

**Comment by LuRsT** [comment]
Thanks for the clear explanation on how you replicated it @yanzhang0219

**Comment by schnej7** [comment]
For anyone looking to speed up fzf further, check out this post: https://dev.to/jerry_schneider/how-i-made-fzf-feel-snappy-again-in-large-repositories-208n
